Rpackage/R/growtree.R

```diff
@@ -0,0 +1,71 @@
+growtree <- structure(function(target.mat, feature.mat, depth=0, maxdepth = Inf,
+                               margin = 0.0, loss="hinge", id = 1L, min_sample = 1,
+                               pred=NULL, side = NULL) {
+
+  #if depth exceeds we stop the further growth of tree
+  if (depth >= maxdepth) return(partynode(id = id, info = pred[[side]]))
+  
+
+  #if sample per node is less than minimum, we stop tree growth
+  if(length(target.mat[, 1]) <= min_sample) return(partynode(id = id, info = pred[[side]]))
+
+  #split the tree at the node.
+  sp <- bestsplit(target.mat, feature.mat, margin, loss,pred)
+  splt <- sp
+
+  #if no split, we stop
+  if (is.null(sp)) return(partynode(id = id, info = pred[[side]]))
+
+  #partysplit object
+  sp <- partysplit(varid = sp$varid, breaks = sp$br)
+
+  #the kids left and right defined
+  kid <- NULL
+  kid$left <- NULL
+  kid$right <- NULL
+  tar <- NULL
+  tar$left <- NULL
+  tar$right <- NULL
+
+  #splitting into kids
+  kidids <- kidids_split(sp, data = feature.mat) #list of right left kids
+
+  #creating partynode list of kids
+  kids <- vector(mode = "list", length = max(kidids, na.rm = TRUE))
+  
+  #if no left right kids, stop tree growth
+  if(length(kids) <= 1) return(partynode(id = id, info = pred[[side]]))
+
+  #creating list of data into
+  for(i in 1:length(feature.mat[,1])){
```

Comment:
Torsten says that this should be changed to subsetting instead of creating a copy of the data. That is, we treat the data as constant and always pass it to the splitting function, while using one subset vector per child, that tells us which examples are in which child. 

It's basically the same idea as what you did here, but you avoid creating a copy of the data.

Reply:
yeah! working on it